packages/core-app-api/src/apis/implementations/StorageApi/WebStorage.ts

```diff
@@ -89,9 +89,19 @@ export class WebStorage implements StorageApi {
   observe$<T extends JsonValue>(
     key: string,
   ): Observable<StorageValueSnapshot<T>> {
+    this.addStorageEventListener();
```

Comment:
So this will add a new subscription every single time that anybody observes for changes, which probably is excessive.

We could reduce it to happening at most once per WebStorage instance, or even better ... exactly once ever, globally, and then iterating through the buckets collection to learn which one(s) to notify when events come in.

Reply:
@freben 

> So this will add a new subscription every single time that anybody observes for changes, which probably is excessive.

I checked the number of storage events listeners inside browser Dev Tools and there were 2 (since there are 2 `WebStorage` observers) and I thought that wouldn't be a big deal.  But I agree with that there should be only one listener as the number of observers will probably increase over time.

> We could reduce it to happening at most once per WebStorage instance, or even better ... exactly once ever, globally, and then iterating through the buckets collection to learn which one(s) to notify when events come in.

I've updated it to be exactly one global event listener for all storage buckets as you suggested.

Thank you for your great suggestion!

Reply:
`@backstage/cli  packages/cli/src/commands/versions/bump.test.ts` test is failing after I

Reply:
> You should just hold a state flag (either in the root, or better, as a private static initialized = false in the class)

That's great, thanks!

Check this out https://github.com/backstage/backstage/pull/15040/commits/5aa0dc604f772d3b825d057b3917f498b6965405

- There's still only one listener.
- Listener will be added only if there's at least one `WebStorage` API subscriber as there's no point in adding a listener if there're no subscribers.